ext/mysqlnd/mysqlnd_ps_codec.c

```diff
@@ -241,7 +242,21 @@ ps_fetch_time(zval * zv, const MYSQLND_FIELD * const field, const unsigned int p
 		t.time_type = MYSQLND_TIMESTAMP_TIME;
 	}
 
-	length = mnd_sprintf(&value, 0, "%s%02u:%02u:%02u", (t.neg ? "-" : ""), t.hour, t.minute, t.second);
+    if (field->decimals > 0 && field->decimals < 7) {
+        length = mnd_sprintf(
+            &value,
+            0,
+            "%s%02u:%02u:%02u.%*u",
```

Comment:
The asterisk is also unnecessary.

Reply:
Sorry but you seem to have missed my previous reply.
When you have 9 as a long, with a decimals value of 3, that is represented as a decimal 9000 (microseconds), but you still want to return 009, not 9.
That being said, I am unsure as of what to add in the place of asterisks if I remove them.

Reply:
You're right. Forgot about the opposite case. Thet it looks like it's missing a zero, should be `%0*u` so a small number is padded on the left. Please lets test this case by inserting the appropriate data, too.